back/engines/commercial/public_api/app/finders/public_api/projects_finder.rb

```diff
@@ -34,5 +41,16 @@ def filter_by_publication_status(scope)
         .joins(:admin_publication)
         .where(admin_publication: { publication_status: @publication_status })
     end
+
+    # Select only the projects that have all the specified topics.
+    def filter_by_topic_ids(scope)
+      return scope unless @topic_ids
+
+      scope
+        .joins(:topics)
+        .where(topics: { id: @topic_ids })
+        .group('projects.id')
+        .having('COUNT(topics.id) = ?', @topic_ids.size)
```

Comment:
Not too sure about this - Why do we need this `having` clause?

Reply:
Ah, OK I see from the tests that this is so that the query only returns results where all topics are present - can't decide if this is the best behaviour or not, but I guess it matches with the principle of the other filters where as you add filters it is an AND not an OR

Reply:
Another reason I chose an AND is because you can simulate the OR logic by sending multiple requests. Implementing one's own AND logic is a bit less straightforward. However, we may still consider adding OR filters later on if it's deemed worthwhile.

Reply:
Yeah think it's good so long as it's clear in the docs. Implementing OR becomes a whole lot more complex